front/src/modules/timesStops/TimesStops.tsx

```diff
@@ -186,18 +209,21 @@ const TimesStops = ({ pathProperties, pathSteps = [] }: TimesStopsProps) => {
         }
       }}
       stickyRightColumn={{
-        component: ({ rowData }) =>
+        component: ({ rowData, rowIndex }) =>
           createDeleteViaButton({
             removeVia: () => removeVia(rowData),
-            isRowVia: isVia(compact(pathSteps), rowData),
+            isRowVia:
+              rowIndex !== 0 &&
+              rowIndex !== pathProperties.allWaypoints?.length - 1 &&
+              isVia(compact(pathSteps), rowData, true),
           }),
       }}
       lockRows
       height={600}
       rowClassName={({ rowData, rowIndex }) =>
         rowIndex === 0 ||
-        rowIndex === pathProperties.allVias.length - 1 ||
-        isVia(compact(pathSteps), rowData)
+        rowIndex === pathProperties.allWaypoints.length - 1 ||
+        isVia(compact(pathSteps), rowData, true)
```

Comment:
As you have changed the type of pathSteps, you can remove all the "compact" of the file I think
```suggestion
        isVia(pathSteps, rowData, true)
```

Reply:
Good catch ! Done.